applications/GeoMechanicsApplication/custom_strategies/builder_and_solvers/residualbased_block_builder_and_solver_linear_elastic_dynamic.h

```diff
@@ -0,0 +1,735 @@
+          mGamma(Gamma),
+          mCalculateInitialSecondDerivative(CalculateInitialSecondDerivative)
+    {
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverLinearElasticDynamic() override = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void InitializeSolutionStep(ModelPart& rModelPart, TSystemMatrixType& rA, TSystemVectorType& rDx, TSystemVectorType& rb) override
+    {
+        BaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
+
+        mPreviousOutOfBalanceVector = TSystemVectorType(BaseType::mEquationSystemSize, 0.0);
+        mCurrentOutOfBalanceVector  = TSystemVectorType(BaseType::mEquationSystemSize, 0.0);
+
+        if (mPreviousExternalForceVector.size() == 0) {
+            mPreviousExternalForceVector = TSystemVectorType(BaseType::mEquationSystemSize, 0.0);
+        }
+    }
+
+    void Build(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart, TSystemMatrixType& rA, TSystemVectorType& rb) override
+    {
+        Timer::Start("Build");
+        this->BuildLHS(pScheme, rModelPart, rA);
+        this->BuildRHSNoDirichlet(pScheme, rModelPart, rb);
+
+        Timer::Stop("Build");
+
+        // apply dirichlet conditions
+        TSystemVectorType dummy_b(rA.size1(), 0.0);
+        TSystemVectorType dummy_rDx(rA.size1(), 0.0);
+
+        BaseType::ApplyDirichletConditions(pScheme, rModelPart, rA, dummy_rDx, rb);
+        BaseType::ApplyDirichletConditions(pScheme, rModelPart, mMassMatrix, dummy_rDx, dummy_b);
+        BaseType::ApplyDirichletConditions(pScheme, rModelPart, mDampingMatrix, dummy_rDx, dummy_b);
+
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverLinearElasticDynamic", BaseType::GetEchoLevel() == 3)
+            << "Before the solution of the system"
+            << "\nSystem Matrix = " << rA << "\nUnknowns vector = " << dummy_rDx
+            << "\nRHS vector = " << rb << std::endl;
+
+        if (mCalculateInitialSecondDerivative) {
+            this->CalculateInitialSecondDerivative(rModelPart, rA, rb);
+            mCopyExternalForceVector = true;
+        }
+
+        // only add dynamics to lhs after calculating intial force vector
+        this->AddDynamicsToLhs(rA, rModelPart);
+
+        // Initialize the linear solver, such that the solver can factorize the matrices already. In case the matrices can be pre-factorized, this step is not
+        // performed during calculation.
+        BaseType::mpLinearSystemSolver->InitializeSolutionStep(rA, dummy_rDx, rb);
+    }
+
+    /**
+     * @brief Function to perform the build of the LHS, mass matrix and damping matrix
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     */
+    void BuildLHS(typename TSchemeType::Pointer pScheme, ModelPart& rModelPart, TSystemMatrixType& rA) override
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR_IF(!pScheme) << "No scheme provided!" << std::endl;
```

Comment:
You could also use 
```suggestion
        KRATOS_ERROR_IF_NOT(pScheme) << "No scheme provided!" << std::endl;
```
Which is slightly more readable in my opinion